### PR TITLE
feat: always use `sha1` crate for sha1-hmac

### PR DESCRIPTION
**Pull request by thomaseizinger**
Currently, `openssl` is used to perform SHA1-HMAC on Windows. The module's comment states that `sha1-asm` does not work on Windows which is true. It would only work on the `windows-gnu` toolchain but that one breaks `openssl`. If https://github.com/RustCrypto/asm-hashes/issues/45 were resolved, we could also compile `sha1` asm on `windows-msvc`.

Simply disabling the `asm` feature on Windows seems to be the best trade-off. This makes things slower on Windows.

As part of this refactoring, I designed the new `sha1-hmac` functionality to be stateless which ended up deleting the `RtpHmac` trait. This trait was only implemented on one type which suggests that it wasn't a very good use of a `trait` anyway.

This is the first step in making the `openssl` dependency optional as discussed in https://github.com/algesten/str0m/issues/413.

### CODE REVIEW

**Comment by thomaseizinger** [comment]
> Simply disabling the `asm` feature on Windows seems to be the best trade-off. This makes things slower on Windows.

@algesten Is that acceptable? I think the proper fix for this performance regression would be to do https://github.com/RustCrypto/asm-hashes/issues/45 for the `sha1` crate. I have not written assembly code before, so perhaps somebody with more experience on that front could tackle this and restore the performance after merging this?

I am also open to other suggestions but I think we will have to stop using `openssl` for `sha1-hmac` in order to make it entirely optional.

File: src/rtp/srtp.rs
Comment (algesten) [inline src/rtp/srtp.rs:149]: How about calling it just `key`, so we can avoid all these destructures to format vertically?

**Review by algesten — APPROVED** [review]
Thanks!

Overall LGTM. My only issue is a code style one with all these vertical formattings.

**Comment by algesten** [comment]
@xnorpx since performance on Windows is very important to you guys, can you weigh in here?

File: src/rtp/srtp.rs
Comment (thomaseizinger) [reply src/rtp/srtp.rs:149]: Okay!

**Comment by xnorpx** [comment]
@algesten we are currently only running str0m on Linux servers, while we would like to prototype clients on Windows is more long term. 

Even so, not having the asm optimizations would not be a blocker to get running and compiling and the cargo.toml comments point that there is optimization improvement if the issue is resolved.

So I think it's fine to remove asm for now on Windows and unblock for the feature work of making openssl a feature.

**Comment by algesten** [comment]
Great! Thanks!

File: src/crypto.rs
Comment (xnorpx) [inline src/crypto.rs:6]: this unwrap will never hit?

Comment (thomaseizinger) [reply src/crypto.rs:6]: The same `unwrap` was present in the previous `From` impl for `Sha1`. It is an internal API, let me see what we can do about it!

Comment (algesten) [reply src/crypto.rs:6]: It's ok to unwrap internally, but there should be a comment of why it is fine.

Comment (thomaseizinger) [reply src/crypto.rs:6]: Interesting, I think this can be hit if the provided STUN password is not 20 bytes long. @algesten Any input on this?

Comment (thomaseizinger) [reply src/crypto.rs:6]: No, `Hmac` normalizes the key to 20 bytes: https://github.com/RustCrypto/MACs/blob/1a674e4c162222468b13c8b87265f4eda52b6531/hmac/src/lib.rs#L113.

**Comment by thomaseizinger** [comment]
> @algesten we are currently only running str0m on Linux servers, while we would like to prototype clients on Windows it is more long term.

FWIW, we will be running it on Windows but I don't think SHA1 performance is going to be a bottleneck for us next to Wireguard! We are also only using ICE from str0m so only sending a few STUN messages to establish a connection.

**Comment by thomaseizinger** [comment]
Awesome, thank you both for the quick review turnaround! :)

**Review by algesten — APPROVED** [review]
Looks great. Let's merge!

**Pull request merged**
